### PR TITLE
Address lookup should fail gracefully

### PR DESCRIPTION
**Pull request by autotelik**
Check current situation/specs
Ensure following
Log error
Return empty address list
Notify user to use Manual entry with specific message
Auto open the manual entry form

### CODE REVIEW

File: spec/services/address_lookup_spec.rb [hunk 1]
```diff
@@ -86,6 +92,81 @@ module AddressLookup
             end
           end
         end
+
+        describe "service down", fails: true do
+
+          before(:each) do
```
Comment (Cruikshanks) [inline spec/services/address_lookup_spec.rb]: Just querying if a ```before``` and ```after``` that do the same thing is duplication?

File: spec/services/address_lookup_spec.rb [hunk 2]
```diff
@@ -86,6 +92,81 @@ module AddressLookup
...
+            VCR.use_cassette("adaptor_no_such_server_uprn") do
+              setup_bad_server
+
+              expect {
+                Adapters::AddressFacade.find_by_uprn("77138")
+              }.to raise_error DigitalServicesCore::AddressServiceNA
+            end
+          end
+
+          it "raises an exception when service cannot be reached" do
```
Comment (Cruikshanks) [inline spec/services/address_lookup_spec.rb]: Like you've done for the config tests below, could we tweak the ```its``` statement to note whether its searching by postcode or uprn.

File: spec/shared_examples/shared_examples_for_address.rb [hunk 3]
```diff
@@ -48,11 +55,48 @@
...
+                               text: I18n.t("errors.messages.digital_services_core.no_address_service_for_postcode"))
+
+      expect(page).to have_css(manual_address_summary_id, text: I18n.t("shared.address.want_to_manaully_add"))
+    end
+  end
+end
+
+
 RSpec.shared_examples "manual_address_entry_valid_and_invalid" do
   # The manual Address option now seems to only appear after an attempt to enter a
```
Comment (Cruikshanks) [inline spec/shared_examples/shared_examples_for_address.rb]: Would you mind rewording the comment to *The manual address option only appears after an attempt to enter a valid postcode. This is by design to help improve the quality of the data we collect.* It's nothing major its just the current wording makes it sound like the current behaviour is unexpected.

File: spec/services/address_lookup_spec.rb (hunk 1 above)
Comment (autotelik) [inline spec/services/address_lookup_spec.rb]: moved duplicated code into helper method but yes both required to ensure config always clean before each test and left clean after context for future tests

**Comment by autotelik** [comment]
For dev and test now uses
 
address_facade_server: <%= ENV['ADDRESS_FACADE_TEST_SERVER'] %>
address_facade_port: <%= ENV['ADDRESS_FACADE_TEST_PORT'] %>

And added 2 new ones to secrets.yml in frontend 

ADDRESS_FACADE_CLIENT_ID='id'
ADDRESS_FACADE_KEY = 'key'

These have been added to env.example

**Comment by rickclare** [comment]
@Cruikshanks did you have that discussion with @autotelik  about ensuring that we use the same environment-variables on Digital-Services-core and Waste-exemptions?  

As a temporary measure to get Address-lookups working in AWS, I have already implemented configurable Address-Facade settings as part of this PR https://github.com/EnvironmentAgency/waste-exemptions/pull/110/files  I have passed those ENV-variable names to the infrastructure team (who have configured them in the AWS ansible provisioning config).

We need to make sure that the environment variables names being used are the same as those that the Infrastructure team are using, as it involves a lot of pain and agro to get them to change config.  (You may have already thought about and solved all this, but just wanted to make sure).

**Pull request merged**